src/ModelContextProtocol.Core/Server/McpServerExtensions.cs

```diff
@@ -234,6 +250,192 @@ public static ValueTask<ElicitResult> ElicitAsync(
+        }
+
+        var jsonElement = AIJsonUtilities.CreateJsonSchema(underlyingType, serializerOptions: serializerOptions);
+
+        if (!TryValidateElicitationPrimitiveSchema(jsonElement, originalType, out var error))
+        {
+            throw new McpException(error);
+        }
+
+        var primitiveSchemaDefinition =
+            jsonElement.Deserialize(McpJsonUtilities.JsonContext.Default.PrimitiveSchemaDefinition);
+        
+        if (primitiveSchemaDefinition is null)
+            throw new McpException($"Type '{originalType.FullName}' is not a supported property type for elicitation requests.");
+
+        return primitiveSchemaDefinition;
+    }
+
+    /// <summary>
+    /// Validate the produced schema strictly to the subset we support. We only accept an object schema
+    /// with a supported primitive type keyword and no additional unsupported keywords.Reject things like
+    /// {}, 'true', or schemas that include unrelated keywords(e.g.items, properties, patternProperties, etc.).
+    /// </summary>
+    /// <param name="schema">The schema to validate.</param>
+    /// <param name="type">The type of the schema being validated, just for reporting errors.</param>
+    /// <param name="error">The error message, if validation fails.</param>
+    /// <returns></returns>
+    private static bool TryValidateElicitationPrimitiveSchema(JsonElement schema, Type type,
+        [NotNullWhen(false)] out string? error)
+    {
+        if (schema.ValueKind is not JsonValueKind.Object)
+        {
+            error = $"Schema generated for type '{type.FullName}' is invalid: expected an object schema.";
+            return false;
+        }
+
+        if (!schema.TryGetProperty("type", out JsonElement typeProperty)
+            || !(typeProperty.ValueKind is JsonValueKind.String or JsonValueKind.Array))
+        {
+            error = $"Schema generated for type '{type.FullName}' is invalid: missing or invalid 'type' keyword.";
+            return false;
+        }
+
+        string? typeKeyword;
+        if (typeProperty.ValueKind == JsonValueKind.String)
+        {
+            typeKeyword = typeProperty.GetString();
+        }
+        else
+        {
+            error = $"Schema generated for type '{type.FullName}' is invalid: unsupported 'type' array.";
+            return false;
+        }
+
+        if (string.IsNullOrEmpty(typeKeyword))
+        {
+            error = $"Schema generated for type '{type.FullName}' is invalid: empty 'type' value.";
+            return false;
+        }
+
+        if (typeKeyword is not ("string" or "number" or "integer" or "boolean"))
+        {
+            error = $"Schema generated for type '{type.FullName}' is invalid: unsupported primitive type '{typeKeyword}'.";
+            return false;
+        }
+
+        if (typeKeyword == "integer")
+            typeKeyword = "number";
```

Comment:
Please avoid using branching without braces.

Reply:
Also, this renaming of the type keyword seems a bit confusing. Couldn't you just add another entry for `integer`? You can reuse the same underlying HashSet if memory use is a concern.